src/utils/api/apiCalls.test.js

```diff
@@ -45,6 +45,243 @@ describe('API', () => {
 
       await expect(API.fetchTitleScroll()).rejects.toEqual(expectedError)
     })
+  })
+
+  describe('getCategoryData', () => {
+    let mockPeople
+
+    beforeEach(() => {
+      mockPeople = [
+        {name: 'Luke', species: 'human'},
+        {name: 'C3PO', species: 'droid'},
+        {name: 'Obi-Wan', species: 'human'}
+      ]
+      window.fetch = jest.fn().mockImplementation(() => {
+        return Promise.resolve({
+          ok: true,
+          json: () => Promise.resolve(mockPeople)
+        })
+      })
+    })
+
+    it('should call fetch with the correct params', async () => {
+      const expected = 'https://swapi.co/api/people'
+
+      await API.getCategoryData('people')
+
+      expect(window.fetch).toHaveBeenCalledWith(expected)
+    })
+
+    it('should return a JSON object if successful', async () => {
+      const expected = [
+        {name: 'Luke', species: 'human'},
+        {name: 'C3PO', species: 'droid'},
+        {name: 'Obi-Wan', species: 'human'}
+      ]
+
+      const result = await API.getCategoryData('people')
+
+      expect(result).toEqual(expected)
+    })
+
+    it('should throw and error if unsuccessful', async () => {
+      const expectedError = Error('Internal Server Error')
+      window.fetch = jest.fn().mockImplementation(() => {
+        return Promise.resolve({
+          ok: false,
+          json: () => Promise.resolve(mockPeople)
+        })
+      })
+
+      expect(API.getCategoryData('people')).rejects.toEqual(expectedError)
+    })
+  })
+
+  describe('getPeople', () => {
+    let mockPeople
+
+    beforeEach(() => {
+      mockPeople = [{
+        name: 'Luke',
+        species: 'https://swapi.co/api/species/1',
+        homeworld: 'https://swapi.co/api/planets/1',
+        height: 30,
+        mass: 100
+      }]
+      window.fetch = jest.fn().mockImplementation(() => {
+        return Promise.resolve({
+          ok: true,
+          json: () => Promise.resolve(mockPeople)
+        })
+      })
+    })
+
+    it('should call fetch with the correct params', async () => {
+      const expected1 = 'https://swapi.co/api/planets/1'
+      const expected2 = 'https://swapi.co/api/species/1'
+
+      await API.getPeople(mockPeople)
+
+      expect(window.fetch).toHaveBeenNthCalledWith(1, expected1)
+      expect(window.fetch).toHaveBeenNthCalledWith(2, expected2)
+    })
+
+    it('should return the correct data if successful', async () => {
+      mockPeople = [{
+        name: 'Luke',
+        species: 'species',
+        homeworld: 'home',
```

Comment:
This is clever.  Like how you replaced the link with 'home', and then mocked out fetch so that if homeworld is 'home' the promise resolves